### PR TITLE
[admission-policy-engine] bootstrap_handler hook checks CRD's statuses

### PR DESCRIPTION
**Pull request by miklezzzz**
## Why do we need it, and what problem does it solve?
Before these changes, the bootstrap handler hook only checked if required Constraint Templates were created in the cluster. 
It appeared it wasn't enough to say that the admission-policy-engine module was bootstrapped as in some cases, like gatekeeper controller manager failure, (or even more sophisticated issues like rego policy syntax errors, etc) Constraint Templates may exist, but resulting CRDs aren't created from them.

## Why do we need it in the patch release (if we do)?
It fixes the module release issue when CRD's can't be created from constraint templates for some reason and helm upgrade fails.

## What is the expected result?
The module upgrade process doesn't fail if the module isn't bootstrapped correctly.

## Checklist
- [x] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: admission-policy-engine
type: fix
summary: Improve bootstrap handler logic.
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @miklezzzz. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @miklezzzz. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @miklezzzz. I've started the workflow for label 'e2e/run/yandex-cloud'...

File: modules/015-admission-policy-engine/hooks/bootstrap_handler.go [hunk 1]
```diff
@@ -17,25 +17,31 @@ limitations under the License.
...
-	"github.com/deckhouse/deckhouse/go_lib/set"
 )
 
 const (
 	pattern = "*.yaml"
 )
 
-// Before creating Gatekepeer's constraints, we have to have running gatekeeper-controller-manager deployment for handling ConstraintTemplates and all required CRDs (constraint templates) for them
-// so, based on ready deployment replicas and constraints templates in place we set the `bootstrapped` flag and create constraints only when true
+type CTemplate struct {
```
Comment (yalosev) [inline modules/015-admission-policy-engine/hooks/bootstrap_handler.go:37]: I think we can make it unexportable, like `cTemplate`

File: modules/015-admission-policy-engine/hooks/bootstrap_handler.go [hunk 2]
```diff
@@ -57,31 +63,78 @@ var _ = sdk.RegisterFunc(&go_hook.HookConfig{
 
 func handleGatekeeperBootstrap(input *go_hook.HookInput) error {
 	var bootstrapped bool
-	var existingTemplates = set.NewFromSnapshot(input.Snapshots["gatekeeper_templates"])
+	templates := input.Snapshots["gatekeeper_templates"]
 
-	if existingTemplates.Size() != 0 {
+	if len(templates) != 0 {
+		existingTemplates := make(map[string]struct {
```
Comment (yalosev) [inline modules/015-admission-policy-engine/hooks/bootstrap_handler.go:69]: you can reuse `cTemplate` here, not to create a new struct, just don't fill the `Name` field

**Review by yalosev — APPROVED** [review]

**Pull request merged**